qutebrowser/browser/browsertab.py

```diff
@@ -133,12 +134,97 @@ class TabData:
     fullscreen = attr.ib(False)  # type: bool
     netrc_used = attr.ib(False)  # type: bool
     input_mode = attr.ib(usertypes.KeyMode.normal)  # type: usertypes.KeyMode
+    splitter = attr.ib(None)
 
     def should_show_icon(self) -> bool:
         return (config.val.tabs.favicons.show == 'always' or
                 config.val.tabs.favicons.show == 'pinned' and self.pinned)
 
 
+class InspectorSplitter(QSplitter):
+
+    """Allows putting an inspector inside the tab.
+
+    Attributes:
+        _main_idx: index of the main webview widget
+        _position: position of the inspector (right/left/top/bottom)
+        _preferred_size: the preferred size of the inpector widget in pixels
+    """
+
+    _main_idx = 0
+    _position = None
+    _preferred_size = None
+
+    def __init__(self, main_webview):
+        super().__init__()
+        self.addWidget(main_webview)
+        self.splitterMoved.connect(self._onSplitterMoved)
+
+    def set_inspector(self, inspector, position):
+        """Set the position of the inspector."""
+        assert position in ['right', 'left', 'top', 'bottom']
+        self._main_idx = 0 if position in ['right', 'bottom'] else 1
+        self.setStretchFactor(self._main_idx, 1)
+        self.setStretchFactor(self._inspector_idx(), 0)
+        self.setOrientation(Qt.Horizontal if position in ['right', 'left']
+                            else Qt.Vertical)
+        self.insertWidget(self._inspector_idx(), inspector)
+        self._position = position
+        self._load_preferred_size()
+        self._adjust_size()
+
+    def _save_preferred_size(self):
+        """Save the preferred size of the inspector widget."""
+        configfiles.state['geometry']['inspector_' + self._position] = str(
```

Comment:
instead of pre-pending inspector, why not just have `inspector_{right,left,top,bottom,window}` as the values to pass in?

Reply:
I just took the same member variable name `_position` as in `inspector.py` so I don't have to think of anything new an used the same payload for consistency.
I don't care too much about this as long as it works.